File: tournaments/reversedSumOfDigits/reversedSumOfDigits.cs
string reversedSumOfDigits(int p, int n) {
    if ((p == 0 && n == 1)) return "0";
    if (9 * n < p || p == 0) return "-1";
    
    var result = "";
    for (int i = 0; i < n-1; i++){
        int tmp = (p - 1) > 9 ? 9: (p - 1);
        p -= tmp;
        result = tmp + result;
    }   
    return p + "" + result;
}

File: tournaments/reversedSumOfDigits/reversedSumOfDigits.js
function reversedSumOfDigits(p, n) {
    if (!p && 1 === n) return '0';
    if (9 * n < p || !p) return '-1';

    const result = [];
    for (let i = 0; n - 1 > i; ++i) {
        const a = 9 < (p - 1) ? 9 : (p - 1);
        p -= a;
        result.unshift(a);
    }
    return p + result.join('');
}
